tests/tests_principal.yml

```diff
@@ -0,0 +1,61 @@
+---
+- name: Test issuing certificate with principal.
+  hosts: all
+  vars:
+    certificate_requests:
+      - name: mycert
+        dns: www.example.com
+        principal: HTTP/www.example.com@EXAMPLE.COM
+        ca: self-sign
+  roles:
+    - linux-system-roles.certificate
+
+- name: Verify certificate
+  hosts: all
+  vars:
+    certificates:
+      - path: /etc/pki/tls/certs/mycert.crt
+        key_path: /etc/pki/tls/private/mycert.key
+        subject:
+          - name: commonName
+            oid: 2.5.4.3
+            value: www.example.com
+        subject_alt_name:
+          - name: DNS
+            value: www.example.com
+          - name: Universal Principal Name (UPN)
+            value: HTTP/www.example.com@EXAMPLE.COM
+            oid: 1.3.6.1.4.1.311.20.2.3
+          - name: Kerberos principalname
+            value: HTTP/www.example.com@EXAMPLE.COM
+            oid: 1.3.6.1.5.2.2
+  tasks:
+    - name: Verify each certificate
+      include_tasks: tasks/assert_certificate_parameters.yml
+      loop: "{{ certificates }}"
+      loop_control:
+        loop_var: cert
+
+
+- name: Test issuing certificate with invalid principal.
```

Comment:
I think I would use a different file for this test. Other then that it looks good.

Reply:
Added changes